src/serialization/serialize-assets.js

```diff
@@ -16,10 +16,14 @@ const serializeAssets = function (runtime, assetType, optTargetId) {
         const currAssets = currTarget.sprite[assetType];
         for (let j = 0; j < currAssets.length; j++) {
             const currAsset = currAssets[j];
-            const asset = currAsset.asset;
+            const asset = currAsset.broken ? currAsset.broken.asset : currAsset.asset;
+            if (!asset) {
```

Comment:
Oh I think I have a bug here, this should actually be:
```suggestion
            if (!(asset && asset.data)) {
```
so that we don't serialize zero byte files... (e.g. if the original costume was missing, it should still be missing, it shouldn't have an empty file in the re-saved .sb3 in its place)

Reply:
Though actually I'll just write this in the affirmative so that it's not confusing

e.g.
```
if (asset && asset.data) {
    assetDescs.push(....);
}
```

Reply:
Okay! Instead of the above change, I made LLK/scratch-storage#388 to handle some inconsistencies and bugs in the way storage resolves assets (especially missing assets). That should eliminate the need for the above change I am talking about to add `&& asset.data`

Reply:
Storage 2.0.0 and 2.0.1 bring in these fixes